sympy/solvers/tests/test_decompogen.py

```diff
@@ -16,6 +16,12 @@ def test_decompogen():
     assert decompogen(Abs(cos(y)**2 + 3*cos(x) - 4), x) == [Abs(x), 3*x + cos(y)**2 - 4, cos(x)]
 
 
+def test_expo():
+    assert decompogen(2**x, x) == [2**x]
+    assert decompogen(2**x - 10, x) == [2**x - 10]
+    assert decompogen(2**(x+3) + 4, x) == [2**(x+3) + 4]
```

Comment:
Shouldn't this decompose x + 3?

Reply:
I wasn't sure whether exponential equations should be decomposed or not. So I thought for any exponential function it should return the equation itself for now.

Reply:
The purpose of decompogen is to assist the solver functions because solving `F(x)` is easier if you can rewrite `F = f o g`, because you then only need to invert `f` and `g` separately. Exponentials equations can be inverted if they are in the right form (with logarithms, or LambertW), so I think it would help. 

For instance, to solve `2**(x + 3) - 4 = 0` you would write it as `2**a - 4 = 0` where `a = x + 3`, giving `a = 2`, and thus `2 = x + 3`, so `x = -1`.